src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostAssignability.scala

```diff
@@ -197,23 +197,26 @@ trait GhostAssignability {
   }
 
   /* ghost type of the callee itself (not considering its arguments or results) */
-  private[separation] def calleeGhostTyping(call: ap.FunctionCall): GhostType = call.callee match {
-    case p: ap.Function => ghost(p.symb.ghost)
-    case p: ap.ReceivedMethod => ghost(p.symb.ghost)
-    case p: ap.MethodExpr => ghost(p.symb.ghost)
-    case _: ap.PredicateKind => GhostType.isGhost
-    case _: ap.DomainFunction => GhostType.isGhost
-    case p: ap.BuiltInFunction => ghost(p.symb.ghost)
-    case p: ap.BuiltInReceivedMethod => ghost(p.symb.ghost)
-    case p: ap.BuiltInMethodExpr => ghost(p.symb.ghost)
-    case _ => GhostType.isGhost // conservative choice
+  private[separation] def calleeGhostTyping(call: ap.FunctionCall): GhostType = {
+    val isExplicitlyGhost = call.callee match {
+      case p: ap.Function => p.symb.ghost
+      case p: ap.ReceivedMethod => p.symb.ghost
+      case p: ap.MethodExpr => p.symb.ghost
+      case _: ap.PredicateKind => true
+      case _: ap.DomainFunction => true
+      case p: ap.BuiltInFunction => p.symb.ghost
+      case p: ap.BuiltInReceivedMethod => p.symb.ghost
+      case p: ap.BuiltInMethodExpr => p.symb.ghost
+      case _ => true // conservative choice
```

Comment:
maybe we should have a `ghost` field either for the class `FunctionCall` or for the trait `FunctionKind` instead of having these big case distinctions that we might forget to update every time we update the codebase.

Reply:
In addition, we should add a case for `ap.Closure`